### PR TITLE
Cancel incomplete lunaria preview builds on new commits

### PR DESCRIPTION
**Pull request by Fryuni**
#### Description

Lunaria continues to run after new commits, which wastes a bit of time but, more importantly, may cause a race in the Lunaria preview.

Concurrency options are the same as the CI file:
https://github.com/withastro/docs/blob/b5dbd5323918b4d45b96d73ccc2b49e6276862fc/.github/workflows/ci.yml#L10-L13

### CODE REVIEW

**Review by lorenzolewis — APPROVED** [review]
LGTM 👍

File: .github/workflows/lunaria.yml [hunk 1]
```diff
@@ -6,6 +6,11 @@ on:
     types: [opened, synchronize]
     branches: [main]
 
+# Automatically cancel in-progress actions on the same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request_target' && github.head_ref || github.ref }}
+  cancel-in-progress: true
+
```
Comment (TheOtterlord) [inline .github/workflows/lunaria.yml:13]:
```suggestion


```

File: .github/workflows/lunaria.yml [hunk 2]
```diff
@@ -6,6 +6,12 @@ on:
...
+
```
Comment (TheOtterlord) [inline .github/workflows/lunaria.yml:14]:
```suggestion

```

**Review by TheOtterlord — COMMENTED** [review]
This doesn't seem to cancel the first run
image

**Comment by Fryuni** [comment]
`pull_request_target` triggers workflows based on the latest commit of the pull request's base branch. So it won't use the fix until it is merged 🙃

**Review by TheOtterlord — APPROVED** [review]
Ahh ok, cool. LGTM then! I'll add this to the merge queue, and will merge after T&D

**Pull request merged**